shared/lib_battery_lifetime.h

```diff
@@ -25,234 +24,78 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-    /// Bilinear interpolation, given the depth-of-discharge and cycle number, return the capacity percent
-    double bilinear(double DOD, int cycle_number);
-
-    std::shared_ptr<cycle_state> state;
-    std::shared_ptr<lifetime_params> params;
-
-private:
-    void initialize();
-
-    friend class lifetime_t;
-};
-
-/*
-Lifetime calendar model
-*/
-
-struct calendar_state {
-    double q_relative_calendar;            // %
-    int day_age_of_battery;
-    double dq_relative_calendar_old;       // (0 - 1)
-
-    friend std::ostream &operator<<(std::ostream &os, const calendar_state &p);
-
-    bool operator==(const calendar_state &p);
-};
-
-class lifetime_calendar_t {
-public:
-    /// Constructors for independent models, owning its state and params
-    lifetime_calendar_t(double dt_hour, const util::matrix_t<double>& calendar_matrix);
-
-    explicit lifetime_calendar_t(double dt_hour, double q0= 1.02, double a= 2.66e-3, double b= -7280, double c= 930);
-
-    /// Constructor as lifetime_t component
-    explicit lifetime_calendar_t(std::shared_ptr<lifetime_params> params_ptr);
+    enum MODEL_CHOICE {
+        CALCYC,
+        NMCNREL         // K. Smith: Life Prediction model coefficients
+    } model_choice;
 
-    lifetime_calendar_t(const lifetime_calendar_t &rhs);
-
-    lifetime_calendar_t &operator=(const lifetime_calendar_t &rhs);
-
-    lifetime_calendar_t *clone();
-
-    /// Given the index of the simulation, the tempertature and SOC, return the effective capacity percent
-    double runLifetimeCalendarModel(size_t lifetimeIndex, double T, double SOC);
-
-    /// Reset or augment the capacity
-    void replaceBattery(double replacement_percent);
-
-    /// Return the relative capacity percentage of nominal (%)
-    double capacity_percent();
-
-    calendar_state get_state();
-
-protected:
-    void runLithiumIonModel(double temp_C, double SOC);
+    std::shared_ptr<calendar_cycle_params> cal_cyc;
 
-    void runTableModel();
+    lifetime_params();
 
-    double dt_day;
+    lifetime_params &operator=(const lifetime_params &rhs);
 
-    std::shared_ptr<calendar_state> state;
-    std::shared_ptr<lifetime_params> params;
-
-private:
-    void initialize();
-
-    friend class lifetime_t;
+    friend std::ostream &operator<<(std::ostream &os, const lifetime_params &p);
 };
 
-/*
-Class to encapsulate multiple lifetime models, and linearly combined the associated degradation and handle replacements
-*/
+struct cycle_state;
+struct calendar_state;
+struct lifetime_nmc_state;
 
 struct lifetime_state {
     double q_relative;                      // total lifetime relative capacity %
+    int n_cycles;
+    double range;
+    double average_range;
+    double day_age_of_battery;
 
-    std::shared_ptr<cycle_state> cycle;
+    // CALCYC model state
     std::shared_ptr<calendar_state> calendar;
+    std::shared_ptr<cycle_state> cycle;
+
+    // NREL NMC model state
+    std::shared_ptr<lifetime_nmc_state> nmc_state;
```

Comment:
Do you mind renaming this so it doesn't have `state` in the name? I think that'll make it more consistent with the naming in the other state structs.